pkgs/applications/window-managers/hyprland/default.nix

```diff
@@ -0,0 +1,71 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, libdrm
+, libinput
+, libxcb
+, libxkbcommon
+, mesa
+, pango
+, wayland
+, wayland-protocols
+, wlroots
+, xcbutilwm
+}:
+
+stdenv.mkDerivation rec {
+  pname = "hyprland";
+  version = "unstable-23-04-2022";
+
+  src = fetchFromGitHub {
+    owner = "vaxerski";
+    repo = pname;
+    rev = "fa38dfd416b664033a96bb4af764fd54f7257d7b";
+    sha256 = "sha256-wP0I8Y2cC6XadL5+XEXJnIKlPDo6u7gJys0TykqBe4A=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+
+    wayland
+  ];
+
+  buildInputs = [
+    libdrm
+    libinput
+    libxcb
+    libxkbcommon
+    mesa
+    pango
+    wayland-protocols
+    wlroots
+    xcbutilwm
+  ];
+
+  postPatch = ''
+    make config
+  '';
+
+  postBuild = ''
+    pushd ../hyprctl
+    g++ -std=c++20 -w ./main.cpp -o ./hyprctl
+    popd
```

Comment:
I haven't tested it but it looks like we can use `make all` instead?: https://github.com/vaxerski/Hyprland/blob/f09de365d5214ea73e6578bd7a72e6c16740b6a0/Makefile#L93 (the Makefile could be improved in general - just in case you're interested to send patches upstream).

Reply:
I didn't want to use `make all` here because it hardcodes `-j 10` and would also pull in ninja as a build dependency